gadgets/src/is_zero.rs

```diff
@@ -63,27 +72,19 @@ impl<F: FieldExt> IsZeroChip<F> {
         value_inv: Column<Advice>,
     ) -> IsZeroConfig<F> {
         // dummy initialization
-        let mut is_zero_expression = Expression::Constant(F::zero());
+        let mut is_zero_expression = 0.expr();
 
         meta.create_gate("is_zero gate", |meta| {
             let q_enable = q_enable(meta);
 
             let value_inv = meta.query_advice(value_inv, Rotation::cur());
             let value = value(meta);
 
-            let one = Expression::Constant(F::one());
-            is_zero_expression = one - value.clone() * value_inv.clone();
+            is_zero_expression = 1.expr() - value.clone() * value_inv;
 
             // This checks `value_inv ≡ value.invert()` when `value` is not
             // zero: value ⋅ (1 - value ⋅ value_inv)
-            let poly1 = value * is_zero_expression.clone();
-            // This checks `value_inv ≡ 0` when `value` is zero:
-            // value_inv ⋅ (1 - value ⋅ value_inv)
-            let poly2 = value_inv * is_zero_expression.clone();
-
-            [poly1, poly2]
-                .into_iter()
-                .map(move |poly| q_enable.clone() * poly)
+            [q_enable * value * is_zero_expression.clone()]
```

Comment:
better add more comments here:

1. if value == 0, is_zero_expression of course is 0
2. if value != 0, require  is_zero_expression == 0, so value_inv should be value.invert()